bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/converter/type/NumericTypeConverter.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.siemenshvac.internal.converter.type;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.siemenshvac.internal.constants.SiemensHvacBindingConstants;
+import org.openhab.binding.siemenshvac.internal.converter.ConverterException;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.types.Type;
+
+import com.google.gson.JsonElement;
+
+/**
+ * Converts between a SiemensHvac datapoint value and an openHAB DecimalType.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+public class NumericTypeConverter extends AbstractTypeConverter {
+    @Override
+    protected boolean toBindingValidation(Type type) {
+        return true;
+    }
+
+    @Override
+    protected @Nullable Object toBinding(Type type, ChannelType tp) throws ConverterException {
+        Object valUpdate = null;
+
+        if (type instanceof QuantityType quantityType) {
+            Number num = (quantityType);
+            valUpdate = num.doubleValue();
```

Comment:
Is this the converter used for temperatures? It seems that units are stripped here, is that intentional? I suspect that temperatures are supported based on the README examples and this `SiemensHvacMetadataRegistryImpl.getCategory`:
```java
        } else if (dptUnit.contains("°C")) {
```

Reply:
Yes, it is used for temperature.
I'm just handling it as a number as we already attach the unit to the meta data description of the ChannelType (see screenshoot below).

So i'm not sure we need to handle it one more time at this place, even if the API returns the unit as a separate field when reading the value.

![image](https://github.com/openhab/openhab-addons/assets/4678815/deb8c563-a8f9-4e57-b313-80ed8981f72b)

![image](https://github.com/openhab/openhab-addons/assets/4678815/35d701de-b013-4d3f-8df6-84eba6cb7997)

![image](https://github.com/openhab/openhab-addons/assets/4678815/55589b55-42ee-48eb-b7df-ccdd08790f36)

Reply:
For [Uom](https://www.openhab.org/docs/concepts/units-of-measurement.html) support you would need to add dimension, i.e. `Number:Temperature`. This will allow the framework to convert to any temperature measurement unit. If you link a `Number:Temperature` item with unit °F to the `Number` channel without dimension, the result will be 14 °F for the example above, which is not expected. The binding needs to inform the framework about the unit.

For a `Number:Temperature` channel, you can update with `QuantityType` rather than `DecimalType`. Since the API actually carries the unit, you can even supply this directly when creating the `QuantityType`. Then the API could even start serving °F, and your item above would still be updated with °C (with correct conversion).